src/nvme/mi-mctp.c

```diff
@@ -213,15 +213,13 @@ static int nvme_mi_mctp_submit(struct nvme_mi_ep *ep,
 		 * somewhere in the header buf */
 		resp->hdr_len = len - sizeof(mic);
 		resp->data_len = 0;
-		memcpy(&mic, ((uint8_t *)resp->hdr) + resp->hdr_len,
-		       sizeof(mic));
+		mic = *(__le32 *)(((uint8_t *)resp->hdr) + resp->hdr_len);
```

Comment:
Hmm, not particular a big fan of these casts and load strategy. E.g. can we sure that the address is aligned?

Reply:
Good question - the header itself *should* be aligned, and we know that the `hdr_len` is 4-byte aligned too, so we should get an aligned result from the add.

However, these loads are all added by a single (small) patch - I'm happy to drop that one if you'd prefer to keep the `memcpy`.

Reply:
Well, I am just one of those guys who get bitten very often by alignment issues on embedded systems. So any direct load is always kind of a danger zone. It's your call, I am fine if you want to use the direct load. Just be aware, I will not resist saying 'I told you so' if we ever happen to hit this problem :)

Reply:
I'll defer to your experience and drop that change :grin:

v2 coming shortly.